mcafee-agent/elx/map.jinja

```diff
@@ -2,7 +2,8 @@
 {%- load_yaml as defaults %}
 
 client_in_ports:
-  - 8591
+  - ''
```

Comment:
This will create a list with one item, an empty string. And the state is looping over the items in the list. So I'd expect this to resolve to this state, which I don't think is desired:

```
Allow ePO Mgmt Inbound Port ''-{{ zone }}:
  module.run:
    - name: firewalld.add_port
    - zone: {{ zone }}
    - port: ''/tcp
    - permanent: True
    - require_in:
      - module: Reload firewalld for McAfee Inbound Port ''
```

Instead, define an actual empty list:

```
client_in_ports: []
```

Reply:
I'm actually not sure I'm comfortable pulling out the default value at this time, since implementations may be relying on that. If we remove it, then that will break implementations until it is specified in the user's pillar...

Reply:
Greh... Shouldn't have been relying on defaults.

Reply:
Ripping out the default still makes it a breaking change, which makes this much harder to merge. If we really want to remove the value, we first have to check every implementation to ensure it is set through pillar. Or we have to implement a deprecation route with a log warning, leave the default for now, and follow up again at some future point in time to remove it.

Reply:
Ok... I can see _maybe_ delaying this and adding a deprecation-warning to the project-README (and set up an appropriate calendar entry to do the actual removal). That said, I'm wondering about the degree of concern, here:

1. The project is not externally-forked
1. The project is not even starred (internally or externally)
1. My understanding was that this particular state isn't what one might call part of the "core" watchmaker functionality. That is, watchmaker would only run this formula if a given user's salt content-archive includes direction to do so?

I mean, I can understand having concern for third-parties that are using it as a standalone, directly-executed module. However, points 1 & 2 would seem to belie that based on our other projects' starring/forking by external users. Similarly, the current default-value isn't even one of the values [specified by McAfee](https://kc.mcafee.com/corporate/index?page=content&id=KB66797). Which is to say, if any external third-parties _are_ currently using this state, it seems a low-order risk that they would be letting this formula drive their port-selection (i.e., relying on the default). Presumably, they'd have deployment needs defined organizationally rather than letting some random project on GitHub drive their port-selection (since the latter would be a "tail wags the dog" from a tool-selection/usage standpoint). Basically, it isn't like we're offering an Apache or Nginx formula that we're removing 80/443 as default ports from (indeed, the already-included _outbound_ ports are staying and are reflective of McAfee's (previously linked-to) documentation. 

Side note: when I do a couple quick Google searches for port 8591 in relation to McAfee and/or HBSS, the only relevant hits returned point to this project. Not seeing vendor documentation or other public documentation suggesting - let alone dictating - "use port 8591". Like I say, while conceivably a breaking-change, everything suggests there would be an extremely low-order risk associated with actual impact from that change.

Reply:
Partly just good project custondianship. We provided and supported a default. Removing it is a breaking change (especially as implemented, since it results in an invalid command).

Partly that we have several implementations for one customer in different environments. Not sure without inspecting each one whether we've overridden the default.

Partly that we have further implementations with a different customer that neither you nor I directly support. We have no insight at this moment into if or how they are using this particular formula.

Partly that this change is also not directly relevant to the feature added or bug being fixed in the PR.

Partly that there doesn't seem to be much/any benefit to be gained by removing the default at this point.

Reply:
> Partly that there doesn't seem to be much/any benefit to be gained by removing the default at this point.

Been bitten, more than once, having to explain to accreditors "why does the host firewall have this exception configured that goes to nothing". Remember, accreditors are the same people that require explanation when you don't have `iptables6` and IPv6-related `sysctl` settings in place when IPv6 has been wholly disabled. I'd rather avoid multiple iterations of explaining the spurious exception.

At any rate, my intention was to fix the `''` to `[]` once this conversation was concluded. Mostly needed to sort out "am I making that fix _now_ or am I posting a pending-deprecation notice (and scheduling said deprecation)".

Reply:
I'd at least recommend moving this change to another PR, so we can get the necessary changes into a watchmaker patch release. Any breaking stuff will end up in a future watchmaker minor release.

Reply:
No worries. I'll revert the change and open a new PR.

Reply:
Reverted in 220222c